File: 3620-maximum-number-of-distinct-elements-after-operations/3620-maximum-number-of-distinct-elements-after-operations.java
// LeetCode: class Solution { public int maxDistinctElements(int[] nums, int k) { ... } }
import java.util.*;
class Solution {
    public int maxDistinctElements(int[] nums, int k) {
        Arrays.sort(nums);
        long occupied = Long.MIN_VALUE / 4L;
        int ans = 0;
        for (int num : nums) {
            if (occupied < (long)num + k) {
                long take = Math.max(occupied + 1, (long)num - k);
                occupied = take;
                ++ans;
            }
        }
        return ans;
    }
}

File: 3620-maximum-number-of-distinct-elements-after-operations/3620-maximum-number-of-distinct-elements-after-operations.py
# LeetCode: class Solution:
#              def maxDistinctElements(self, nums: List[int], k: int) -> int:
from typing import List
import math

class Solution:
    def maxDistinctElements(self, nums: List[int], k: int) -> int:
        nums.sort()
        occupied = -10**30
        ans = 0
        for num in nums:
            if occupied < num + k:
                take = max(occupied + 1, num - k)
                occupied = take
                ans += 1
        return ans
